dartagnan/src/main/java/com/dat3m/dartagnan/program/Thread.java

```diff
@@ -78,7 +79,7 @@ public Register getRegister(String name){
 
     public Register addRegister(String name, int precision){
         if(registers.containsKey(name)){
-            throw new RuntimeException("Register " + id + ":" + name + " already exists");
+            throw new AlreadyExistentVariableException("Register " + id + ":" + name + " already exists");
```

Comment:
I would probably go for either `IllegalStateException` or a custom exception that is a little more general, something like `DuplicateException` or `AlreadyPresentException`.

Reply:
If this call gets in, it would result in a malformed program, thus I'll go with `MalformedProgramException`